### PR TITLE
[AIRFLOW-2832] Lint and resolve inconsistencies in Markdown files

### PR DESCRIPTION
**Pull request by tedmiston**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2832
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

- Inspired by other recent issues related to linter errors in Python and JS (AIRFLOW-2783, AIRFLOW-2800, AIRFLOW-2803)
- This PR does a few things:
  - Resolves linter errors in markdown files across the project (ignores errors that aren't super useful on GitHub such as line wrapping and putting `<urls>` in brackets)
  - Clarifies that commit message length of 50 characters doesn't include the Jira issue tag
  - Replaces usage of JIRA with Jira the way it's styled nowadays by [Atlassian](https://www.atlassian.com/software/jira) and [Wikipedia](https://en.wikipedia.org/wiki/Jira_(software))
  - Makes code block formatting consistent

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

The changes in this PR are restricted to linting documentation.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

n/a

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by tedmiston** [comment]
This PR is now squashed and ready for review.

I'm not sure that there's any one best person to review these changes but in a git log, I see that @bolkedebruin, @Fokko, and @r39132 have modified some of these files in recent history.

File: dev/README.md [hunk 1]
```diff
@@ -72,25 +76,33 @@ origin	https://github.com/<USER>/airflow (push)
...
+
+*Note: `compile_assets` command build the frontend assets (JS and CSS) files for the
 Web UI using webpack and npm. Please make sure you have `npm` installed on your local machine globally.
 Details on how to install `npm` can be found in CONTRIBUTING.md file.*
 
 After that navigate to relative directory i.e., `cd dist` and sign the release files.
 
-`../dev/sign.sh <the_created_tar_ball.tar.gz` 
+```bash
+../dev/sign.sh <the_created_tar_ball.tar.gz
```
Comment (Fokko) [inline dev/README.md:105]: If I would go fully nitpick mode, I would ask you to add a space after the `<`

**Review by Fokko — APPROVED** [review]
LGTM, Thanks @tedmiston for picking this up!

**Comment by tedmiston** [comment]
@Fokko Thanks for the quick merge!  I'll make a note to look into linting the bash code in Airflow and see if we have enough for a PR there.

**Pull request merged**